packages/@lwc/engine-core/src/framework/reporting.ts

```diff
@@ -9,12 +9,13 @@ import { VM } from './vm';
 
 export const enum ReportingEventId {
     CrossRootAriaInSyntheticShadow = 0,
+    CompilerRuntimeVersionMismatch = 1,
 }
 
 type ReportingDispatcher = (
     reportingEventId: ReportingEventId,
-    tagName: string,
-    vmIndex: number
+    tagName?: string,
+    vmIndex?: number
```

Comment:
It would be nice to know the tagName/VM, but at the point where we freeze the template, we don't know what that is. In fact, the component hasn't even been instantiated or inserted into the DOM yet.

I figure reporting something is better than nothing.

Reply:
It might be useful to know what version the component was originally compiled with for this event; what were your thoughts on event-specific payloads?

O11y requires a well defined schema so we would still have to statically define the structure of each event's metric somewhere.

Reply:
I didn't want implementation details of o11y to bleed over into LWC OSS.

That said, we could have some kind of `payload` object where the TypeScript type depends on the `ReportingEventId` passed in. We could even move `tagName` and `vmIndex` into this payload.

I'd say: let's work on getting these 3 PRs in, and then do the payload schema in a future PR. Otherwise it will be hard to refactor everything across these PRs 😅

Reply:
>It would be nice to know the tagName/VM 

@nolanlawson I was planning to add the `tagName` to the `registerComponent` calls as part of the dynamic components PR.  I can also add it to [`registerTemplate`](https://github.com/salesforce/lwc/blob/81b528476921ac7a474544432cbda78e80725d63/packages/@lwc/engine-core/src/framework/secure-template.ts#L27). 

Do you think adding the `tagName` to `registerTemplate` would be useful for this?

Reply:
@jmsjtu Yes, that would be useful. Once both PRs are in, we can look into reporting the `tagName`.